packages/kernel/src/worker-runtime.ts

```diff
@@ -54,6 +54,21 @@ script_runner.moduleAutoLoadPromise = __moduleAutoLoadPromise__
 `);
 }
 
+async function setEnv(
+  pyodide: PyodideInterface,
+  env: Record<string, string>,
+): Promise<void> {
+  // This function is used to set environment variables both during initialization
+  // and during the runtime.
+  const envJSON = JSON.stringify(env);
```

Comment:
Can you try something like this? (I didn't test this snippet actually)
```ts
const os = pyodide.pyimport("os")
os.environ.update(env)
```
This may treats the Object-to-Dict conversion more safely and can avoids defining the temp variable `env_dict`.

Reply:
Agreed! This is much better! Ended up being
```
const os = pyodide.pyimport("os");
os.environ.update(pyodide.toPy(env));
```

Question is if we want the API to be `updateEnv` rather than `setEnv`.